src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -385,6 +413,23 @@ string CreateUniqueRemoteName(IRepository repo, string name)
             return uniqueName;
         }
 
+        async Task<string> ExtractToTempFile(IRepository repo, string commitSha, string fileName, Encoding encoding)
+        {
+            var contents = await gitClient.ExtractFile(repo, commitSha, fileName) ?? string.Empty;
+            return CreateTempFile(fileName, commitSha, contents, encoding);
+        }
+
+        static string CreateTempFile(string fileName, string commitSha, string contents, Encoding encoding)
+        {
+            var tempDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var tempFileName = $"{Path.GetFileNameWithoutExtension(fileName)}@{commitSha}{Path.GetExtension(fileName)}";
```

Comment:
We shouldn't be trusting that the filename coming from the PR is valid under the current locale and encoding or even valid to Windows. There are certain filenames that can be created in other OSs that the Windows fs can't deal with, and there are certain characters and encodings that might not match the local. In this case, ignoring the string formatting warning is not a good option, and using the original filename to create a temp file is probably not a good option either.

We're already creating a GUID in the directory name, if we use the GUID instead for the filename we avoid both issues entirely. Alternatively, if you want to reuse the cached temp file, you could encode the filename with `Convert.ToBase64String(Encoding.UTF8.GetBytes(filename))` and put it in a known temp directory.